From 9b0d74db531fbe858572b954687aeec7cf5dce16 Mon Sep 17 00:00:00 2001
From: Potin Lai <potin.lai@quantatw.com>
Date: Sat, 2 May 2026 22:53:06 +0800
Subject: [PATCH] Fix NVMeMI sensor build error

Add necessary changes to fix NVMe MI sensor build error with upstream
codebase.
---
 src/nvmemi/NVMeMiSensor.hpp       |  2 +-
 src/nvmemi/NVMeMiSensorMain.cpp   |  8 +++---
 src/nvmemi/NVMeMiStatusSensor.cpp | 41 +++++++++++++++++--------------
 3 files changed, 27 insertions(+), 24 deletions(-)

diff --git a/src/nvmemi/NVMeMiSensor.hpp b/src/nvmemi/NVMeMiSensor.hpp
index cea7974..de85bda 100644
--- a/src/nvmemi/NVMeMiSensor.hpp
+++ b/src/nvmemi/NVMeMiSensor.hpp
@@ -32,7 +32,7 @@
 class NVMeSensor : public Sensor
 {
   public:
-    static constexpr const char* sensorType = "NVME1000";
+    static constexpr const char* sensorType = "NVMeMI";
 
     NVMeSensor(sdbusplus::asio::object_server& objectServer,
                boost::asio::io_context& io,
diff --git a/src/nvmemi/NVMeMiSensorMain.cpp b/src/nvmemi/NVMeMiSensorMain.cpp
index db46737..8fb62dd 100644
--- a/src/nvmemi/NVMeMiSensorMain.cpp
+++ b/src/nvmemi/NVMeMiSensorMain.cpp
@@ -46,7 +46,7 @@ extern "C"
 #include <sdbusplus/bus/match.hpp>
 #include <sdbusplus/message.hpp>
 #include <sdbusplus/message/native_types.hpp>
-#include <tal.hpp>
+//#include <tal.hpp>
 
 #include <algorithm>
 #include <array>
@@ -514,7 +514,7 @@ void createSensors(boost::asio::io_context& io,
             // Clear flag when sensor configuration handling completes
             createSensorsInProgress = false;
         });
-    getter->getConfiguration(std::vector<std::string>{
+    getter->getConfiguration(std::vector<std::string_view>{
         NVMeSensor::sensorType, NVMeStatusSensor::sensorType});
 }
 
@@ -671,7 +671,7 @@ int main()
 
     boost::asio::io_context io;
     auto systemBus = std::make_shared<sdbusplus::asio::connection>(io);
-    systemBus->request_name("xyz.openbmc_project.NVMeSensor");
+    systemBus->request_name("xyz.openbmc_project.NVMeMISensor");
     sdbusplus::asio::object_server objectServer(systemBus, true);
     objectServer.add_manager("/xyz/openbmc_project/sensors");
 
@@ -734,7 +734,7 @@ int main()
     std::vector<std::unique_ptr<sdbusplus::bus::match_t>> matches =
         setupPropertiesChangedMatches(
             *systemBus,
-            std::to_array<const char*>(
+            std::to_array<const std::string_view, 2>(
                 {NVMeSensor::sensorType, NVMeStatusSensor::sensorType}),
             eventHandler);
 
diff --git a/src/nvmemi/NVMeMiStatusSensor.cpp b/src/nvmemi/NVMeMiStatusSensor.cpp
index 9bd071a..f1fca90 100644
--- a/src/nvmemi/NVMeMiStatusSensor.cpp
+++ b/src/nvmemi/NVMeMiStatusSensor.cpp
@@ -39,6 +39,9 @@
 
 namespace fs = std::filesystem;
 
+using AssociationList =
+    std::vector<std::tuple<std::string, std::string, std::string>>;
+
 NVMeStatusSensor::NVMeStatusSensor(
     sdbusplus::asio::object_server& objectServer,
     boost::asio::io_context& /*unused*/,
@@ -82,10 +85,10 @@ NVMeStatusSensor::NVMeStatusSensor(
     sdbusplus::xyz::openbmc_project::Inventory::server::Item::present(true);
     sdbusplus::xyz::openbmc_project::State::Decorator::server::
         OperationalStatus::functional(false);
-    sdbusplus::xyz::openbmc_project::State::Decorator::server::
-        OperationalStatus::state(
-            sdbusplus::xyz::openbmc_project::State::Decorator::server::
-                OperationalStatus::StateType::None);
+    // sdbusplus::xyz::openbmc_project::State::Decorator::server::
+    //     OperationalStatus::state(
+    //         sdbusplus::xyz::openbmc_project::State::Decorator::server::
+    //             OperationalStatus::StateType::None);
 }
 
 NVMeStatusSensor::~NVMeStatusSensor()
@@ -105,7 +108,7 @@ bool NVMeStatusSensor::sample()
     return scanDelay == 0;
 }
 
-void NVMeStatusSensor::updateStatus(bool present, bool functional, bool fault)
+void NVMeStatusSensor::updateStatus(bool present, bool functional, [[maybe_unused]] bool fault)
 {
     // Update present status
     sdbusplus::xyz::openbmc_project::Inventory::server::Item::present(present);
@@ -115,18 +118,18 @@ void NVMeStatusSensor::updateStatus(bool present, bool functional, bool fault)
         OperationalStatus::functional(functional);
 
     // Update fault status
-    if (fault)
-    {
-        sdbusplus::xyz::openbmc_project::State::Decorator::server::
-            OperationalStatus::state(
-                sdbusplus::xyz::openbmc_project::State::Decorator::server::
-                    OperationalStatus::StateType::Fault);
-    }
-    else
-    {
-        sdbusplus::xyz::openbmc_project::State::Decorator::server::
-            OperationalStatus::state(
-                sdbusplus::xyz::openbmc_project::State::Decorator::server::
-                    OperationalStatus::StateType::None);
-    }
+    // if (fault)
+    // {
+    //     sdbusplus::xyz::openbmc_project::State::Decorator::server::
+    //         OperationalStatus::state(
+    //             sdbusplus::xyz::openbmc_project::State::Decorator::server::
+    //                 OperationalStatus::StateType::Fault);
+    // }
+    // else
+    // {
+    //     sdbusplus::xyz::openbmc_project::State::Decorator::server::
+    //         OperationalStatus::state(
+    //             sdbusplus::xyz::openbmc_project::State::Decorator::server::
+    //                 OperationalStatus::StateType::None);
+    // }
 }
-- 
2.53.0-Meta

